tests/unit/test_network_connections.py

```diff
@@ -4081,6 +4095,119 @@ def test_convert_passwd_flags_nm(self):
+                            "network": "198.51.100.128",
+                            "prefix": 26,
+                            "gateway": "198.51.100.1",
+                            "metric": 2,
+                            "table": 30400,
+                        },
+                    ],
+                },
+            }
+        ]
+        self.validator = network_lsr.argument_validator.ArgValidator_ListConnections()
+        self.rt_parsing = network_lsr.argument_validator.IPRouteUtils()
+        # the connection index is 0 because there is only one connection profile
+        # defined here
+        self.connection_index = 0
+
+    def test_valid_numeric_route_tables(self):
+        """
+        Test that the value between 1 and 4294967295 are the valid value for numeric
+        route tables and the value will not be normalized
+        """
+
+        self.validator.validate_route_tables(
+            self.validator.validate(self.test_connections)[0],
+            self.connection_index,
+        )
+        self.assertEqual(
+            self.test_connections[0]["ip"]["route"][0]["table"],
+            30400,
+        )
+
+        self.test_connections[0]["ip"]["route"][0]["table"] = 200
+        self.validator.validate_route_tables(
+            self.validator.validate(self.test_connections)[0],
+            self.connection_index,
+        )
+        self.assertEqual(
+            self.test_connections[0]["ip"]["route"][0]["table"],
+            200,
+        )
+
+    def test_invalid_numeric_route_tables(self):
+        """
+        Test that the value less than 1 or greater than 4294967295 are the invalid
+        value for numeric route tables
+        """
+
+        self.test_connections[0]["ip"]["route"][0]["table"] = 0
+        val_min = 1
+        val_max = 0xFFFFFFFF
+        self.assertRaisesRegex(
+            ValidationError,
+            "route table value is %s but cannot be less then %s"
+            % (
+                self.test_connections[0]["ip"]["route"][0]["table"],
+                val_min,
+            ),
+            self.validator.validate,
+            self.test_connections,
+        )
+
+        self.test_connections[0]["ip"]["route"][0]["table"] = 4294967296
+        self.assertRaisesRegex(
+            ValidationError,
+            "route table value is %s but cannot be greater then %s"
+            % (
+                self.test_connections[0]["ip"]["route"][0]["table"],
+                val_max,
+            ),
+            self.validator.validate,
+            self.test_connections,
+        )
+
+    def test_validate_route_tables(self):
+        """
+        Test that the `validate_route_tables()` will find the table id mapping from
+        `IPRouteUtils.get_route_tables_mapping()`.
+        """
+
+        self.test_connections[0]["ip"]["route"][0]["table"] = "eth"
+
+        old_getter = (
+            network_lsr.argument_validator.IPRouteUtils.get_route_tables_mapping
+        )
```

Comment:
move the storing and restoring of the old_getter to the `SetUp` and teardown function of the test.